Move import into function to avoid GDAL as a global dependency

mpauly opened the pull request:
Hi, 
this should fix #1631 - The problem arises because in commit 39c2adcd720485410c289601d696473027539358 an import was moved from within a function to the top level in `haystack/fields.py`. The import references `django.contrib.gis.geos`. That import in turn causes a lot of other imports that end in a call to GDAL. This call fails if GDAL is not installed. The fix is simple: Just move the import back to the function. The fix is tested in one of our installations (without GDAL).
Cheers,
Martin

acdha commented:
Thanks — I had started this in https://github.com/acdha/django-haystack/tree/defer-gdal-import which has a few other places where the GIS imports needed to be deferred.

mpauly commented:
Ah, great. I wasn't sure if I should also modify the tests and decided against it. Do you have any clue when you will find time to merge your branch?

acdha commented:
My branch isn't finished yet — it's on the TODO list but I haven't had much time for OSS. If you want to help get it to the point where the tests are passing I'd be happy to merge it.

mpauly commented:
I now moved all the imports from `django.contrib.gis.geos` to the functions in which they are used. The tests seem to be passing. Additionally I updated the two regex, that you updated as well as.

acdha commented:
Thanks for picking this up — I'll look into a point release soon since #1631 had a few people blocked on it
